### ISSUE TITLE
Getting googletest to creating reports with filename-tags

### ISSUE THREAD

**Issue by wenns** [wenns/sonar-cxx#407]
Guys, 

I've found a possibility to $subj. 
You just put an 
```C++
RecordProperty("filename", "src/util/tests/testcache.cc");
```
Into your GTEST, and voila, the reports includes an additional XML attribute with this name and value.
Look here: https://code.google.com/p/googletest/wiki/AdvancedGuide#Logging_Additional_Information
The original poster in this thread is Jorge, by the way ;)

**Comment by jmecosta** [comment]
ive forgot about that already, for some reason i havent use this. instead i
have a post processing task that manages those tags.

i might again take a look at this... any way the important for the post is
you will need to modify the macro definition (TEST and TEST_F) to include
the RecordProperty. having those in the test is not helpful, it all needs
to be manage by the platform... if i manage to make it work i will provide
the source code..

**Comment by wenns** [comment]
Look at 'googletest_project'  if you're curious how I would do it. In our projects we can avoid the constants alltogether, since our buildsystem already uses paths relative to projects root and ```__FILE__``` expands accordingly. Cool.

As for modifying the macro definition: this was my first thought, too. Drawbacks: 
1. You dont catch skipped tests (cause RecordProperty wont be executed)
2. It cannot be done easily since the opening curly bracket ```{``` is **not** part of the macro. To go this way, you have to pull it into the macro definition, but this would make your tests look ugly, as you wont see the matching parantheses, e.g.:
```C++
TEST(fixture, testname)
   code;
}
```

**Comment by jmecosta** [comment]
Not sure, but in our case we need to change the macro. i dont like the idea
to include the record macro in the tests (i was thinking also in the
__FILE__ but used in the macro TEST and TEST_F). since people will forget
about it.

i would really like to change the macro, but as per 1 and 2 it might be a
problem also... will investigate at some point....
